src/mlang/backend_compilers/bir_to_ocaml.ml

```diff
@@ -0,0 +1,383 @@
+  if List.length mppf_stmts = 0 then Format.fprintf oc "%s@]" "()"
+  else Format.fprintf oc "%a@]" (generate_stmts program) mppf_stmts
+
+let generate_mpp_functions (oc : Format.formatter) (program : Bir.program) =
+  let functions =
+    Bir.FunctionMap.bindings
+      (Bir_interface.context_agnostic_mpp_functions program)
+  in
+  let function_names, _ = List.split functions in
+  let pp_print_mpp_functions fmt function_names =
+    Format.pp_print_list ~pp_sep:pp_mpp_function_separator
+      (generate_mpp_function program)
+      fmt function_names
+  in
+  Format.fprintf oc "let rec %a@," pp_print_mpp_functions function_names
+
+let generate_rov_function (program : Bir.program) (oc : Format.formatter)
+    (rov : Bir.rule_or_verif) =
+  let tname, stmts =
+    match rov.rov_code with
+    | Rule stmts -> ("rule", stmts)
+    | Verif stmt -> ("verif", [ stmt ])
+  in
+  Format.fprintf oc "@[<v 1>let m_%s_%s (context : m_context) : unit =@,%a@]"
+    tname (Pos.unmark rov.rov_name) (generate_stmts program) stmts
+
+let generate_rov_functions (oc : Format.formatter) (program : Bir.program) :
+    unit =
+  let rovs = Bir.ROVMap.bindings program.rules_and_verifs in
+  let _, rovs = List.split rovs in
+  Format.pp_print_list ~pp_sep:pp_function_separator
+    (generate_rov_function program)
+    oc rovs
+
+let generate_header (oc : Format.formatter) () : unit =
+  Format.fprintf oc "@[<v 0>open Mvalue@,@]"
+
+let generate_output (oc : Format.formatter)
+    (function_spec : Bir_interface.bir_function) : unit =
+  let output_vars =
+    List.map fst (Bir.VariableMap.bindings function_spec.func_outputs)
+  in
+  let name_and_pos_list var_list =
+    List.map
+      (fun var -> (get_var_pos var, Pos.unmark var.mir_var.name))
+      var_list
+  in
+  let print_line fmt (position, name) =
+    Format.fprintf fmt "{alias = \"%s\" ; value = (Array.get tgv %d).value} ::"
+      name position
+  in
+  let pp_print_output_get fmt output_vars =
+    Format.pp_print_list print_line fmt (name_and_pos_list output_vars)
+  in
+  Format.fprintf oc "let output (tgv : m_array) : output_list =@,%a []@,"
+    pp_print_output_get output_vars
+
+let generate_input_handler (oc : Format.formatter)
+    (function_spec : Bir_interface.bir_function) : unit =
+  let input_vars =
+    List.map fst (Bir.VariableMap.bindings function_spec.func_variable_inputs)
+  in
+  let get_position_and_alias variable : (int * string) option =
+    let tgv_pos = get_var_pos variable in
+    Option.map (fun alias -> (tgv_pos, alias)) (get_var_alias variable)
+  in
+  let pp_print_line fmt variable : unit =
+    match get_position_and_alias variable with
+    | Some (position, alias) ->
+        Format.fprintf fmt
+          "let tgv_positions = TgvPositionMap.add \"%s\" %d tgv_positions in"
+          alias position
+    | None -> ()
+  in
+  let pp_print_position_map fmt input_vars =
+    Format.pp_print_list pp_print_line fmt input_vars
```

Comment:
Using `pp_print_list` with default argument `~pp_sep` set to a cut will leave blank lines in a vertical box, with the expected displeasing result in the generated file.

Reply:
Easily solved by using an empty separator and moving the break hint up in the « Some » member of the match, avoiding to insert a cut resulting in a blank line in the « None » case.